File: 1726_tuple_with_same_product.java
class Solution {
    public int tupleSameProduct(int[] x) {
        Map<Integer, Integer> y = new HashMap<>();
        int z = 0;
        for (int i = 0; i < x.length; i++)
            for (int j = i + 1; j < x.length; j++)
                y.put(x[i] * x[j], y.getOrDefault(x[i] * x[j], 0) + 1);
        for (int v : y.values())
            z += v * (v - 1) * 4;
        return z;
    }
}

File: 1726_tuple_with_same_product.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& x) {
        unordered_map<int, int> y;
        int z = x.size(), w = 0;
        for(int i = 0; i < z; i++) 
            for(int j = i + 1; j < z; j++) 
                y[x[i] * x[j]]++;
        for(auto& q : y) 
            w += q.second * (q.second - 1) * 4;
        return w;
    }
};
